Modules/_csv.c

```diff
@@ -1342,75 +1281,98 @@ csv_writerow(PyObject *op, PyObject *seq)
             quoted = 1;
             break;
         case QUOTE_STRINGS:
-            quoted = PyUnicode_Check(field);
+            quoted = !is_none && PyUnicode_Check(field);
             break;
         case QUOTE_NOTNULL:
-            quoted = field != Py_None;
+            quoted = !is_none;
             break;
-        default:
+        default: /* Default QUOTE_MINIMAL or other to minimal behavior */
             quoted = 0;
             break;
         }
 
-        null_field = (field == Py_None);
-        if (PyUnicode_Check(field)) {
-            append_ok = join_append(self, field, quoted);
-            Py_DECREF(field);
+        if (is_none) {
+            /* None is NULL. */
```

Comment:
And that's why `is_none` should be kept `null_field` so that we can remove this comment.